lib/iex/lib/iex/introspection.ex

```diff
@@ -38,25 +37,28 @@ defmodule IEx.Introspection do
         {mod, fun}
 
       {fun, []} ->
-        {find_decompose_fun(fun), fun}
+        {find_decompose_fun(fun, context), fun}
 
       _ ->
         call
     end
   end
 
-  defp find_decompose_fun(fun) do
-    Enum.find(@default_modules, Kernel, fn mod ->
-      Keyword.has_key?(mod.__info__(:functions), fun) or
-        Keyword.has_key?(mod.__info__(:macros), fun)
+  defp find_decompose_fun(fun, context) do
+    context.functions
+    |> Keyword.merge(context.macros, fn _k, v1, v2 -> v1 ++ v2 end)
```

Comment:
I think this broke special forms because they are not really imported, so they are never listed in `__CALLER__`.

Also, the merge would make this unnecessarily expensive. Maybe we could do something like this:

    find_import(fun, context.functions) || find_import(fun, context.macros) || find_special_form(fun) || Kernel

And something similar for the function below.